src/d/actor/d_a_player_main.cpp

```diff
@@ -3154,12 +5109,138 @@ BOOL daPy_lk_c::procControllWait() {
 
 /* 80112E60-80113044       .text procWait_init__9daPy_lk_cFv */
 BOOL daPy_lk_c::procWait_init() {
-    /* Nonmatching */
+    /* Nonmatching - regalloc, uVar2 */
+    u16 uVar1;
+    s32 uVar2;
+    int iVar3;
+    int iVar4;
+    f32 dVar5;
+
+    if ((mDemo.getDemoMode() == daPy_demo_c::DEMO_UNK2A_e) &&
+        (m_tex_scroll_heap.field_0x6 == 0xFFFF))
+    {
+        changeTextureAnime(LKANM_BTP_TMABACC, LKANM_BTK_TEUR, -1);
+        voiceStart(38);
+        m34D0 = 0x14;
+    }
+    if (mCurProc == daPyProc_WAIT_e) {
+        return false;
+    }
+    if ((((!dComIfGp_event_runCheck() && (mCurProc == daPyProc_FREE_WAIT_e)) &&
+          (mFrameCtrlUnder[UNDER_MOVE0_e].getRate() > 0.01f)) &&
+         (checkNoUpperAnime())))
+    {
+        return false;
+    }
+    iVar3 = checkPlayerGuard();
+    commonProcInit(daPyProc_WAIT_e);
+    mVelocity = 0.0f;
+    iVar4 = checkRestHPAnime();
+    if ((iVar4 != 0) && (iVar3 == 0)) {
+        uVar2 = (mTexAnmIndexTable[daPyFace_TMABAF].mBtpIdx - m_tex_anm_heap.mIdx) >> 5 & 0xFF;
+        uVar1 = m3530;
+        setSingleMoveAnime(ANM_WAITATOB, daPy_HIO_move_c0::m.field_0x68,
+                           daPy_HIO_move_c0::m.field_0x6C, daPy_HIO_move_c0::m.field_0x10,
+                           daPy_HIO_move_c0::m.field_0x70);
+        if (uVar2 == 0) {
+            onModeFlg(ModeFlg_00000400);
+            offModeFlg(ModeFlg_00000100);
+        } else {
+            setTextureAnime(0xe, uVar1);
+        }
+    } else {
+        setBlendMoveAnime(daPy_HIO_basic_c0::m.field_0xC);
+    }
+    current.angle.y = shape_angle.y;
+    mDirection = DIR_NONE;
+    m35A0 = 0.0f;
+    dVar5 = cM_rndF(150.0f);
+    m34D2 = (s16)(300.0f + dVar5);
+    return true;
 }
 
 /* 80113044-801133FC       .text procWait__9daPy_lk_cFv */
 BOOL daPy_lk_c::procWait() {
-    /* Nonmatching */
+    /* Nonmatching - regalloc, uVar3 */
+    s32 uVar3;
+    u16 uVar1;
+    s16 sVar4;
+    dBgS_AcchCir* pdVar5;
+
+    if ((m36A0.abs2XZ() <= 1.0000001e-6f) && (m36AC.abs2XZ() >= 25.0f)) {
```

Comment:
Does this still work? Might just be float imprecision
```suggestion
    if ((m36A0.abs2XZ() <= 0.000001f) && (m36AC.abs2XZ() >= 25.0f)) {
```

Reply:
Doesn't work.